code/modules/virus2/centrifuge.dm

```diff
@@ -42,11 +46,18 @@
 
 	attack_hand(user)
 
+//Also handles luminosity
 /obj/machinery/centrifuge/update_icon()
-	..()
-	if(! (stat & (BROKEN|NOPOWER)) && (isolating || curing))
+	if(stat & BROKEN)
+		icon_state = "[icon_state]b"
```

Comment:
Now I'm wondering if you even tested this (knowing you you didn't).

This will break if the proc gets called twice in a row without it going to the base state in between.